lib/lexer.js

```diff
@@ -26,7 +28,14 @@ module.exports = function Lexer (input) {
 
     // are we infinite looping? throw an error
     if (last === current) {
-      throw new Error(`Cannot parse character "${char}" at ${line}:${col}`)
+      throw new Error({
```

Comment:
Right, but it's overriding a JS default, this is not a good practice. That's like adding methods to native array -- it may be helpful in one particular case, but it's overall discouraged as it changes set language defaults that all developers know and understand. I'd call it `SugarmlError` or something.